lambda/recipes-responder/src/main.ts

```diff
@@ -104,19 +127,57 @@ export async function processRecord({
 	}
 }
 
-export const handler: EventBridgeHandler<string, CrierEvent, void> = async (
-	event,
-) => {
+export const handler: Handler<
+	CrierEventBridgeEvent | ReindexEventBridgeEvent,
+	void
+> = async (event) => {
 	const contentPrefix = getContentPrefix();
 	const staticBucketName = getStaticBucketName();
 	const fastlyApiKey = getFastlyApiKey();
+	const outgoingEventBus = getOutgoingEventBus();
+	const capiBaseUrl = getCapiBaseUrl();
+
+	const updatesTotal = await (async () => {
+		switch (event['detail-type']) {
+			case ContentUpdateEventDetail:
+			case ContentDeleteEventDetail: {
+				return await processRecord({
+					eventDetail: event.detail,
+					staticBucketName,
+					fastlyApiKey,
+					contentPrefix,
+					outgoingEventBus,
+				});
+			}
+			case ReindexEventDetail: {
+				let totalCount = 0;
+
+				console.log(
+					`Received ${
+						event.detail.articleIds.length
+					} articles to reindex: \n${event.detail.articleIds.join(', \n')}`,
+				);
 
-	const updatesTotal = await processRecord({
-		eventDetail: event.detail,
-		staticBucketName,
-		fastlyApiKey,
-		contentPrefix,
-	});
+				for (const articleId of event.detail.articleIds) {
+					totalCount += await handleContentUpdateByCapiUrl({
+						capiUrl: `${capiBaseUrl}/${articleId}`,
+						contentType: ContentType.ARTICLE,
+						staticBucketName,
+						fastlyApiKey,
+						contentPrefix,
+						outgoingEventBus,
+					});
+				}
+
+				console.log(`Reindexed ${event.detail.articleIds.length} articles`);
+				return totalCount;
```

Comment:
very minor niggle - to keep with the coding style used in e.g. `processRecord` I'd suggest putting this into its own function?

Reply:
e5239ef5f1fc460d94dd9aae54af22984a493f18